gpytorch/kernels/scale_kernel.py

```diff
@@ -49,15 +49,18 @@ class ScaleKernel(Kernel):
     """
 
     def __init__(self, base_kernel, batch_size=1, log_outputscale_prior=None, param_transform=torch.exp):
-        super(ScaleKernel, self).__init__(has_lengthscale=False, batch_size=batch_size, param_transform=param_transform)
+        super(ScaleKernel, self).__init__(has_lengthscale=False, batch_size=batch_size)
         self.base_kernel = base_kernel
         self.register_parameter(
-            name="log_outputscale", parameter=torch.nn.Parameter(torch.zeros(batch_size)), prior=log_outputscale_prior
+            name="log_outputscale",
+            parameter=torch.nn.Parameter(torch.zeros(batch_size)),
+            prior=log_outputscale_prior,
+            transform=param_transform,
         )
 
     @property
     def outputscale(self):
-        return self.param_transform(self.log_outputscale)
+        return self.transfom_parameter("log_outputscale", self.log_outputscale)
```

Comment:
This fails with an `AttributeError` even when you do something trivial like the following (I have no idea why): 

```
from gpytorch.kernels import ScaleKernel, RBFKernel

sk = ScaleKernel(RBFKernel())
sk.outputscale
``

Reply:
to clarify: The `AttributeError` is w.r.t to `outputscale`, so it doesn't even enter this function here.

Reply:
lol this is totally silly, you can see the typo inside the @property definition above: `transfom_parameter`. The error message in this case is just totally bogus, why doesn't it throw the AttributeError for the correct issue?

Reply:
also, having the `Property` github handle must be the worst thing (or any name that is a common decorator for that matter)